### PR TITLE
Refactor and use a BatchTokenRequest/Response type

### PR DESCRIPTION
**Pull request by chris-wood**
Based on mailing list discussion and [this PR](https://github.com/ietf-wg-privacypass/base-drafts/pull/508) against the base drafts.

cc @tfpauly

### CODE REVIEW

File: draft-ietf-privacypass-batched-tokens.md [hunk 1]
```diff
@@ -295,13 +328,180 @@ requests per client per key.
...
+* Token Structure: As defined in {{Section 2.2 of AUTHSCHEME}}
+* Token Key Encoding: Serialized using SerializeElement from {{Section 2.1 of OPRF}}
+* TokenChallenge Structure: As defined in {{Section 2.1 of AUTHSCHEME}}
+* Publicly Verifiable: N
+* Public Metadata: N
+* Private Metadata: N
+* Batched Issuance: Y
+* Nk: 32
+* Nid: 32
+* Reference: {{private-flow}}
```
Comment (raphaelrobert) [inline draft-ietf-privacypass-batched-tokens.md:349]:
```suggestion
* Reference: {{ISSUANCE, Section 5}}
```

File: draft-ietf-privacypass-batched-tokens.md [hunk 2]
```diff
@@ -295,13 +328,180 @@ requests per client per key.
...
+
+: N/A
+
+Encoding considerations:
+
+: "binary"
+
+Security considerations:
+
+: see {{security}}
```
Comment (raphaelrobert) [inline draft-ietf-privacypass-batched-tokens.md:387]:
```suggestion
: see {{security-considerations}}
```

File: draft-ietf-privacypass-batched-tokens.md [hunk 3]
```diff
@@ -295,13 +328,180 @@ requests per client per key.
...
```
Comment (raphaelrobert) [inline draft-ietf-privacypass-batched-tokens.md:460]:
```suggestion
: see {{security-considerations}}
```

File: draft-ietf-privacypass-batched-tokens.md [hunk 4]
```diff
@@ -295,13 +328,180 @@ requests per client per key.
...
+
+Change controller:
 
-| Value  | Name                                        | Publicly Verifiable | Public Metadata | Private Metadata | Nk  | Reference        |
-|:-------|:--------------------------------------------|:--------------------|:----------------|:-----------------|:----|:-----------------|
-| 0xF901 | Batched Token VOPRF (P-384, SHA-384) | N                   | N               | N                | 32  | This document    |
-| 0xF91A | Batched Token VOPRF (ristretto255, SHA-512) | N                   | N               | N                | 32  | This document    |
-{: #aeadid-values title="Token Types"}
+: IETF
+{: spacing="compact"}
```
Comment (raphaelrobert) [inline draft-ietf-privacypass-batched-tokens.md:507]:
```suggestion
{: spacing="compact"}

```

**Review by raphaelrobert — APPROVED** [review]
Thanks, looks good to me. Just a few suggestions to unclog the CI. 

Let's merge this once the [upstream PR](https://github.com/ietf-wg-privacypass/base-drafts/pull/508) is merged.

File: draft-ietf-privacypass-batched-tokens.md [hunk 5]
```diff
@@ -44,15 +44,20 @@ multiple instances of the issuance protocol in parallel, e.g., over a
...
+
+The token type for this issuance protocol is the same as in the non-batched
+issuance protocol. This is because the underlying cryptographic protocol is
+the same. As a result, the batched issuance protocol in this document only
+works with token types that support batching.
+
+This batched issuance protocol registers one new token type ({{iana-token-type}}),
 to be used with the PrivateToken HTTP authentication scheme defined in
-{{!AUTHSCHEME=I-D.ietf-privacypass-auth-scheme}}.
+{{!AUTHSCHEME=I-D.ietf-privacypass-auth-scheme}}. It also
```
Comment (chris-wood) [inline draft-ietf-privacypass-batched-tokens.md:60]:
```suggestion
{{!AUTHSCHEME=I-D.ietf-privacypass-auth-scheme}}.
```

File: draft-ietf-privacypass-batched-tokens.md [hunk 6]
```diff
@@ -295,13 +328,180 @@ requests per client per key.
...
+
+* Value: 0xF91A
+* Name: VOPRF (ristretto255, SHA-512)
+* Token Structure: As defined in {{Section 2.2 of AUTHSCHEME}}
+* Token Key Encoding: Serialized using SerializeElement from {{Section 2.1 of OPRF}}
+* TokenChallenge Structure: As defined in {{Section 2.1 of AUTHSCHEME}}
+* Publicly Verifiable: N
+* Public Metadata: N
+* Private Metadata: N
+* Batched Issuance: Y
```
Comment (chris-wood) [inline draft-ietf-privacypass-batched-tokens.md:346]:
```suggestion
```

**Review by tfpauly — APPROVED** [review]
Do we want to explicitly mention how the media type would be applied to other types, like type 2? Or leave that for another PR?

**Pull request merged**